### PR TITLE
NATIVE-92: Loading and storing

### PR DESCRIPTION
**Pull request by NKlug**

### CODE REVIEW

File: src/modules/endpoint/sagas/loadLanguages.js [hunk 1]
```diff
@@ -5,9 +5,17 @@ import { call } from 'redux-saga/effects'
...
 import request from '../request'
 import { baseUrl } from '../constants'
+import MemoryDatabase from '../MemoryDatabase'
 
-export default function * loadLanguages (city: string): Saga<?Array<LanguageModel>> {
+export default function * loadLanguages (city: string, database: MemoryDatabase): Saga<void> {
+  yield call(database.readLanguages)
+
+  if (database.languagesLoaded()) {
+    console.dir(database.languages)
```
Comment (Schedulaar) [inline src/modules/endpoint/sagas/loadLanguages.js]: remove  console.dir

File: src/modules/endpoint/MemoryDatabase.js [hunk 2]
```diff
@@ -130,12 +136,17 @@ class MemoryDatabase {
...
+  eventsLoaded = () => this._events !== null
+
   /**
    * @returns {Promise<void>} which resolves to the number of bytes written or rejects
    */
-  writeCategories (): Promise<number> {
+  writeCategories = async () => {
     if (!this.categoriesMap) {
-      throw new Error('MemoryDatabase does not have data to save!')
+      console.warn('MemoryDatabase does not have data to save!')
```
Comment (Schedulaar) [inline src/modules/endpoint/MemoryDatabase.js]: why did you make these errors warnings?

File: src/modules/endpoint/MemoryDatabase.js [hunk 3]
```diff
@@ -184,29 +195,73 @@ class MemoryDatabase {
...
+  writeEvents = async () => {
+    if (!this._events) {
+      console.warn('MemoryDatabase does not have data to save!')
+      return
+    }
+    const path = this.getContentPath('events')
+    await this.writeFile(path, JSON.stringify(this._events))
+  }
+
+  readResourceCache = async () => {
```
Comment (Schedulaar) [inline src/modules/endpoint/MemoryDatabase.js]: we should not use lambda functions as function declarations if we don't need them.
Instead do sth like
```js
async readResourceCache () {
```

File: src/modules/endpoint/MemoryDatabase.js (hunk 2 above)
Comment (NKlug) [reply src/modules/endpoint/MemoryDatabase.js]: Not really sure. Reverting it.

File: src/modules/endpoint/sagas/loadLanguages.js (hunk 1 above)
Comment (NKlug) [reply src/modules/endpoint/sagas/loadLanguages.js]: Done.

File: src/modules/endpoint/MemoryDatabase.js (hunk 3 above)
Comment (NKlug) [reply src/modules/endpoint/MemoryDatabase.js]: Done.

File: src/modules/endpoint/sagas/fetchResourceCache.js [hunk 4]
```diff
@@ -28,25 +28,29 @@ const createErrorMessage = (fetchResult: FetchResultType) => {
   }, '')
 }
 
-export default function * fetchResourceCache (city: string, language: string, fetchMap: FetchMapType): Saga<ResourceCacheStateType> {
+export default function * fetchResourceCache (city: string, language: string, fetchMap: FetchMapType, database: MemoryDatabase): Saga<void> {
+  yield call(database.readResourceCache)
+
+  if (isEmpty(fetchMap)) {
+    return
```
Comment (maxammann) [inline src/modules/endpoint/sagas/fetchResourceCache.js]: I think we should always fire an action here. In this case a success action should be fired.

Comment (NKlug) [reply src/modules/endpoint/sagas/fetchResourceCache.js]: Done.

**Pull request merged**